src/api/publicationApi.ts

```diff
@@ -1,116 +1,97 @@
 import Axios from 'axios';
 import { Dispatch } from 'redux';
 
-import {
-  createPublicationFailure,
-  createPublicationSuccess,
-  getPublicationFailure,
-  getPublicationSuccess,
-  updatePublicationFailure,
-  updatePublicationSuccess,
-} from '../redux/actions/publicationActions';
 import i18n from '../translations/i18n';
-import { Publication, PublicationFileMap, PublicationMetadata } from '../types/publication.types';
-import { ApiBaseUrl, StatusCode } from '../utils/constants';
+import { Publication, PublicationFileMap, PublicationMetadata, DoiPublication } from '../types/publication.types';
+import { ApiBaseUrl, StatusCode, API_URL, API_TOKEN } from '../utils/constants';
+import { addNotification } from '../redux/actions/notificationActions';
 
-interface DoiPublication {
-  url: string;
-  owner: string;
-}
-export const createNewPublicationFromDoi = (url: string, owner: string) => {
+export const createNewPublicationFromDoi = async (url: string, owner: string, dispatch: Dispatch) => {
   const data: DoiPublication = {
     url,
     owner,
   };
-  return async (dispatch: Dispatch) => {
-    try {
-      const response = await Axios({
-        method: 'POST',
-        url: `/${ApiBaseUrl.PUBLICATIONS}/doi`,
-        data: data,
-        headers: {
-          Accept: 'application/json',
-          'Content-Type': 'aplication/json',
-        },
-      });
-      if (response.status === StatusCode.OK) {
-        dispatch(createPublicationSuccess());
-      } else {
-        dispatch(createPublicationFailure(i18n.t('feedback:error.create_publication')));
-      }
-    } catch {
-      dispatch(createPublicationFailure(i18n.t('feedback:error.create_publication')));
+  try {
+    const response = await Axios.post(`/${ApiBaseUrl.PUBLICATIONS}/doi`, data, {
+      headers: {
+        Authorization: `Bearer ${API_TOKEN}`,
+      },
+    });
+    if (response.status === StatusCode.OK) {
+      dispatch(addNotification(i18n.t('feedback:success.create_publication')));
```

Comment:
vi får vel en del data tilbake her som vi skal populere videre i skjemaet?

Reply:
ja, bare getResource som er testet atm. lagde ny sak NÅ-284 for CRUD resource